app/src/main/java/es/wolfi/passman/API/Credential.java

```diff
@@ -275,6 +335,111 @@ public static Credential fromJSON(JSONObject j, Vault v) throws JSONException {
         return c;
     }
 
+    public void save(Context c, final AsyncHttpResponseHandler responseHandler) {
+        RequestParams params = new RequestParams();
+        params.setUseJsonStreamer(true);
+
+        try {
+            JSONObject icon = new JSONObject();
+            icon.put("type", false);
+            icon.put("content", "");
+
+            params.put("vault_id", getVaultId());
+            params.put("label", label);
+            params.put("description", description);
+            params.put("created", getCreated());
+            params.put("changed", getChanged());
+            params.put("tags", tags);
+            params.put("email", email);
+            params.put("icon", icon);
+            params.put("username", username);
+            params.put("password", password);
+            params.put("url", url);
+            params.put("renew_interval", getRenewInterval());
+            params.put("expire_time", getExpireTime());
+            params.put("delete_time", getDeleteTime());
+            params.put("files", files);
+            params.put("custom_fields", customFields);
+            params.put("otp", otp);
+            params.put("compromised", compromised);
+            params.put("hidden", isHidden());
+
+            requestAPI(c, "credentials", params, "POST", responseHandler);
+        } catch (JSONException | MalformedURLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void update(Context c, final AsyncHttpResponseHandler responseHandler) {
+        RequestParams params = new RequestParams();
+        params.setUseJsonStreamer(true);
+
+        try {
+            JSONObject icon;
+
+            if (favicon == null || favicon.equals("") || favicon.equals("null")) {
+                icon = null;
+            } else {
+                try {
+                    icon = new JSONObject(favicon);
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                    icon = null;
+                }
+            }
+
+            params.put("vault_id", getVaultId());
```

Comment:
Wouldn't be better to have all this logic on a common function?

Reply:
I've already thought about it and it's already on my todo list :)
Comes with the next PR.